content/en/blog/2022-06-13-unikraft-releases-hyperion.md

```diff
@@ -0,0 +1,218 @@
+
+In general the structure of both registration methods is similar.
+The main difference is how the entries and folders are structured in the memory.
+Static entries are ordered in an array saved in a sections table and dynamic entries are held inside folders within memory as an ordered linked list.
+Because of this, searching for static folders is faster, thanks to cache storage locality.
+
+In order to allocate only the space strictly necessary in static registrations, link-time sections are generated for each library that is part of the current build.
+To do this, the build system is invoked to generate a list of all used libraries.
+With this list, headers and linker scripts are generated and are used by the build.
+Each library can easily add new registrations in its own storage and access information stored in others.
+
+#### Usage
+
+To use static registration within your lib `libmylib`, `ukstore` will have to be first included:
+```c
+#include <uk/store.h>
+```
+
+Afterwards you can very simply register your getters & setters via the exposed macro:
+
+```c
+UK_STORE_STATIC_ENTRY(
+  my_registered_entry,
+  type,
+  my_getter,
+  my_setter
+);
+```
+Where:
+- `my_registered_entry` - the name of the ukstore entry (this will be used to access it);
+- `type` - the type of the entry (`s8`, `u8`, `s16`, `u16`, `s32`, `u32`, `s64`, `u64`, `uptr`, `charp`);
+- `my_getter` - the function that returns the metric;
+- `my_setter` - the function that sets a metric.
+
+A function that returns a `u8` will have this signature:
+
+```c
+int my_getter(void *cookie, __u8 *dst);
+```
+
+After this the metric is registered and can be accessed anywhere within the application or the unikernel.
+The cookie is optional and it is used as extra storage for dynamic entries.
+
+For quick access to the registered entries, you can pre-emptively get and save them inside any init function, like so:
+
+You will first need some space to save it:
+```c
+static struct uk_store_entry *my_registered_entry;
+my_registered_entry = uk_store_get_entry(libmylib, NULL, "my_registered_entry");
+```
+Where:
+- `libmylib` - library where you are getting the entry from (you can see the registration of the library within `build/libukstore/include/uk/_store_libs.h` if you are unsure);
+- `NULL` - the folder you are getting the entry from (`NULL` for static entry);
+- `"my_registered_entry"` - the name of the entry.
+
+
+Then, when appropriate, you can retrieve the value:
+
+```c
+uk_store_get_value(my_registered_entry, type, &storage);
+```
+Where:
+- `my_entry` - the entry extracted before;
+- `type` - the type of the storage (can be different from the one in the entry but you have to be careful to check if `-ERANGE` may be returned);
+- `storage` - variable to store the metric returned through `ukstore`.
```

Comment:
```suggestion
- `my_entry` - the entry extracted before;
- `u8` - Actually, the type field for the target variable. The value is automatically up- or down-casted if source and target type do not match. In case the target type is too small to hold the value, `-ERANGE` is returned;
- `&curr_metric` - the actual reference of the target variable to store the result.
```